### ISSUE TITLE
OnOffSwitch needs updating

### ISSUE THREAD

**Issue by mwarnett** [codenameone/CodenameOne#2556]
The OnOffSwitch UI component needs updating. 

It has not kept up with the evolution of on/off switches especially on Android. Attached is an Android 8.0 settings panel which shows two on/off switches, one in each state.

CodenameOne is still generating separate On/Off buttons on Android which take up a lot of space.

![screenshot_20180923-131739](https://user-images.githubusercontent.com/14006739/45932611-9216b880-bf33-11e8-865e-31e78be92e81.png)

**Comment by codenameone** [comment]
I think it warrants a complete rewrite on iOS too. It might make sense to deprecate the current component and create a new one.

**Comment by ramsestom** [comment]
The code of a Material design look and fill (by default) OnOffSwitch that you can use as a starting point for your rewrite of this component:
[MDCOnOffSwitch.zip](https://github.com/codenameone/CodenameOne/files/2415994/MDCOnOffSwitch.zip)

I had to remove and change a few stuff that depend on my own version of CN1 (mostly the support of dp and sp units and the usage of my own system of animations) to make it fully compatible with the official CN1 (tell me if you have import issues, I may have missed some) but it still should be a good starting point for a rewrite of the OnOffSwitch component.

**Comment by shannah** [comment]
@ramsestom Thanks for the component.  I used it as a basis for https://github.com/codenameone/CodenameOne/blob/master/CodenameOne/src/com/codename1/components/Switch.java

The new Switch component supports Android and iOS native themes out of the box, and styling it for other platforms should be significantly easier than the old switch.  The iOS7Theme and android_holo_theme include fairly close replicas of iOS and Android native switches.  I have provided reference styling for the other themes (legacy iOS and Android, and UWP) that don't match their native switches exactly, but still look good.

That javadocs include details on how to customize the rendering of this component using styles and theme constants.
